### PR TITLE
Fix failure when debug_kit connection does not exist 

### PR DESCRIPTION
**Pull request by markstory**
When the host application doesn't have SQLite installed failing to save the toolbar data should not cause the page to 500. Instead we can log the error and not end up showing the toolbar iframe as there is not Request persisted.

Refs cakephp/cakephp#15492

### CODE REVIEW

File: src/ToolbarService.php [hunk 1]
```diff
@@ -254,9 +255,16 @@ public function saveData(ServerRequest $request, ResponseInterface $response)
...
         ];
-        /** @var \DebugKit\Model\Table\RequestsTable $requests */
-        $requests = $this->getTableLocator()->get('DebugKit.Requests');
-        $requests->gc();
+        try {
+            /** @var \DebugKit\Model\Table\RequestsTable $requests */
+            $requests = $this->getTableLocator()->get('DebugKit.Requests');
+            $requests->gc();
+        } catch (MissingDatasourceConfigException $e) {
+            Log::warning('Unable to save request. You need to define the debug_kit connection.');
```
Comment (othercorey) [inline src/ToolbarService.php:263]:
```suggestion
            Log::warning('Unable to save request. Check your custom debut_kit database connection or check that SQLite extension is enabled..');
```

File: src/ToolbarService.php [hunk 2]
```diff
@@ -254,9 +255,19 @@ public function saveData(ServerRequest $request, ResponseInterface $response)
...
-        /** @var \DebugKit\Model\Table\RequestsTable $requests */
-        $requests = $this->getTableLocator()->get('DebugKit.Requests');
-        $requests->gc();
+        try {
+            /** @var \DebugKit\Model\Table\RequestsTable $requests */
+            $requests = $this->getTableLocator()->get('DebugKit.Requests');
+            $requests->gc();
+        } catch (MissingDatasourceConfigException $e) {
+            Log::warning(
+                'Unable to save request. Check your debug_kit datasource connection ' .
```
Comment (ADmad) [inline src/ToolbarService.php:264]: We should also add a big fat note in the readme asking people to check the error logs for details if Debugkit isn't showing up :).

Comment (markstory) [reply src/ToolbarService.php:264]: Done. We can always add more reasons in the future.

File: psalm.xml [hunk 3]
```diff
@@ -7,7 +7,7 @@
     xsi:schemaLocation="https://getpsalm.org/schema/config vendor/vimeo/psalm/config.xsd"
     autoloader="tests/bootstrap.php"
     usePhpDocMethodsWithoutMagicCall="true"
-    errorBaseline="psalm-baseline.xml"
+    errorBaseline="./psalm-baseline.xml"
```
Comment (dereuromark) [inline psalm.xml:10]: Is this really necessary?

**Pull request merged**